src/components/Refresh/Refresh.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { Button, MenuItem, Icon, DropdownButton } from 'patternfly-react';
+
+import { config } from '../../config';
+import { PollIntervalInMs } from '../../types/Common';
+
+type Props = {
+  id: string;
+  pollInterval?: PollIntervalInMs;
+  handleRefresh: () => void;
+  onSelect: (selected: PollIntervalInMs) => void;
+};
+
+type State = {
+  pollInterval?: PollIntervalInMs;
+  pollerRef?: number;
+};
+
+class Refresh extends React.Component<Props, State> {
+  static readonly POLL_INTERVALS = config().toolbar.pollInterval;
+
+  constructor(props: Props) {
+    super(props);
+    let pollerRef: number | undefined = undefined;
+    if (this.props.pollInterval) {
+      pollerRef = window.setInterval(this.props.handleRefresh, this.props.pollInterval);
+    }
+    this.state = {
+      pollInterval: this.props.pollInterval,
+      pollerRef: pollerRef
+    };
+  }
+
+  componentWillUnmount() {
+    if (this.state.pollerRef) {
+      clearInterval(this.state.pollerRef);
+    }
+  }
+
+  updatePollInterval = (pollInterval: PollIntervalInMs) => {
+    let newRefInterval: number | undefined = undefined;
+    if (this.state.pollerRef) {
+      clearInterval(this.state.pollerRef);
+    }
+    if (pollInterval > 0) {
+      newRefInterval = window.setInterval(this.props.handleRefresh, pollInterval);
+    }
+    this.setState({ pollerRef: newRefInterval, pollInterval: pollInterval });
+    this.props.onSelect(pollInterval);
+  };
+
+  render() {
+    if (this.state.pollInterval !== undefined) {
+      return (
+        <>
+          <label style={{ paddingRight: '0.5em', marginLeft: '1.5em' }}>Refreshing</label>
+          <DropdownButton id={this.props.id} title={Refresh.POLL_INTERVALS[this.state.pollInterval]}>
+            {Object.keys(Refresh.POLL_INTERVALS).map(strKey => {
+              const key = Number(strKey);
+              return (
+                <MenuItem
+                  key={key}
+                  eventKey={key}
+                  active={key === this.state.pollInterval}
+                  onSelect={this.updatePollInterval}
+                >
+                  {Refresh.POLL_INTERVALS[key]}
+                </MenuItem>
+              );
+            })}
+          </DropdownButton>
+          <span style={{ paddingLeft: '0.5em' }}>
+            <Button id="refresh_button" onClick={this.props.handleRefresh}>
```

Comment:
Setting an "id" in the component, perhaps could duplicates with the same component in other page ?
In other words, can the component be used in more areas without having this id="refresh_button" defined multiple times ?

Reply:
You're right, I'll change it.

Reply:
done